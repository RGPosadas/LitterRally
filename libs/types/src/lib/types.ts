// Types go here

export interface Message {
  message: string;
  info: { production: boolean };
}

export interface IMedia {
  id: number;
  mediaType: string;
}

/**
 * Restriction options for events
 */
export enum EventRestriction {
  NONE,
  FRIENDS_ONLY,
  INVITE_ONLY,
}
