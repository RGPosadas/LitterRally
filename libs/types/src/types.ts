// Types go here
import { Request, Response } from 'express';
export interface Message {
  message: string;
  info: { production: boolean };
}

export interface IMedia {
  id: number;
  mediaType: string;
}

/**
 * Restriction options for events
 */
export enum EventRestriction {
  NONE,
  FRIENDS_ONLY,
  INVITE_ONLY,
}

export enum RegistrationMethod {
  LOCAL = 'LOCAL',
  GOOGLE = 'GOOGLE',
  FACEBOOK = 'FACEBOOK',
  TWITTER = 'TWITTER',
}

export enum RelationshipType {
  FRIENDS,
  ADDED_ME,
  PENDING_REQUEST,
  NONE,
}

export const EventRestrictionMap = ['Everyone', 'Friends', 'Invite Only'];

export interface IEvent {
  id: number;
  title: string;
  startDate: Date;
  endDate: Date;
  description: string;
  image?: IMedia;
  restriction: EventRestriction;
  participants?: IUser[];
  host?: IUser;
  coHosts?: IUser[];
  location?: ILocation;
}

export interface ISerializedEvent extends Omit<IEvent, 'startDate' | 'endDate'> {
  startDate: string;
  endDate: string;
}

export interface ILocation {
  id?: number;
  title: string;
  event?: IEvent;
  coordinates?: IPoint;
  placeId?: string;
}

export interface IPoint {
  lat: number;
  long: number;
}

export interface IUser {
  id: number;
  name?: string;
}

export interface IMedia {
  id: number;
  mediaType: string;
}

export interface IAuthToken {
  id: number;
  iat: number;
  exp: number;
}

export type GqlContext = {
  req: Request;
  res: Response;
};

export interface IRefreshResponse {
  ok: boolean;
  accessToken: string;
}

export interface IRoute {
  url: string;
  displayName: string;
}

export class IGooglePlace {
  description: string;
  placeId: string;
}

export interface IWebSocketParams {
  authToken: string;
}

export interface IPost {
  id: number;
  user: IUser;
  parentPost: IPost;
  message: string;
  createdTime: Date;
  medias?: IMedia;
  channel: IChannel;
}

export interface ISerializedPost extends Omit<IPost, 'createdTime'> {
  createdTime: string;
}

export interface IChannel {
  id: number;
  name: string;
  rights: number;
  event?: ISerializedEvent;
  participants: IUser[];
  posts: ISerializedPost[];
}
