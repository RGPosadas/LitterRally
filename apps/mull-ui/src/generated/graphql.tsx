import { gql } from '@apollo/client';
import * as Apollo from '@apollo/client';
export type Maybe<T> = T | null;
export type Exact<T extends { [key: string]: unknown }> = { [K in keyof T]: T[K] };
export type MakeOptional<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]?: Maybe<T[SubKey]> };
export type MakeMaybe<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]: Maybe<T[SubKey]> };
/** All built-in and custom scalars, mapped to their actual values */
export type Scalars = {
  ID: string;
  String: string;
  Boolean: boolean;
  Int: number;
  Float: number;
  /** A date-time string at UTC, such as 2019-12-03T09:54:33Z, compliant with the date-time format. */
  DateTime: any;
  /** The `Upload` scalar type represents a file upload. */
  Upload: any;
};

export type CreateEventInput = {
  description: Scalars['String'];
  endDate: Scalars['DateTime'];
  image: MediaInput;
  restriction: Scalars['Int'];
  startDate: Scalars['DateTime'];
  title: Scalars['String'];
};

export type CreateUserInput = {
  dob?: Maybe<Scalars['DateTime']>;
  email: Scalars['String'];
  name: Scalars['String'];
  password?: Maybe<Scalars['String']>;
  registrationMethod: RegistrationMethod;
};

export type Event = {
  __typename?: 'Event';
  description: Scalars['String'];
  endDate: Scalars['DateTime'];
  id: Scalars['ID'];
  image?: Maybe<Media>;
  restriction: Scalars['Float'];
  startDate: Scalars['DateTime'];
  title: Scalars['String'];
};

export type LoginInput = {
  email: Scalars['String'];
  password: Scalars['String'];
};

export type LoginResult = {
  __typename?: 'LoginResult';
  accessToken: Scalars['String'];
};

export type Media = {
  __typename?: 'Media';
  id: Scalars['Float'];
  mediaType: Scalars['String'];
};

export type MediaInput = {
  id: Scalars['Float'];
  mediaType: Scalars['String'];
};

export type Mutation = {
  __typename?: 'Mutation';
  addParticipantToEvent: Scalars['Boolean'];
  createEvent: Event;
  createUser: User;
  deleteEvent: Event;
  deleteUser: User;
  login: LoginResult;
  removeParticipantFromEvent: Scalars['Boolean'];
  updateEvent: Event;
  updateUser: User;
  uploadFile: Media;
};

export type MutationAddParticipantToEventArgs = {
  eventId: Scalars['Float'];
  userId: Scalars['Float'];
};

export type MutationCreateEventArgs = {
  createEventInput: CreateEventInput;
};

export type MutationCreateUserArgs = {
  createUserInput: CreateUserInput;
};

export type MutationDeleteEventArgs = {
  id: Scalars['Int'];
};

export type MutationDeleteUserArgs = {
  id: Scalars['Int'];
};

export type MutationLoginArgs = {
  loginInput: LoginInput;
};

export type MutationRemoveParticipantFromEventArgs = {
  eventId: Scalars['Float'];
  userId: Scalars['Float'];
};

export type MutationUpdateEventArgs = {
  updateEventInput: UpdateEventInput;
};

export type MutationUpdateUserArgs = {
  updateUserInput: UpdateUserInput;
};

export type MutationUploadFileArgs = {
  file: Scalars['Upload'];
};

export type Query = {
  __typename?: 'Query';
  coHostEvents: Array<Event>;
  discoverEvents: Array<Event>;
  event: Event;
  events: Array<Event>;
  hostEvents: Array<Event>;
  participatingEvents: Array<Event>;
  user: User;
  users: Array<User>;
};

export type QueryCoHostEventsArgs = {
  userId: Scalars['Int'];
};

export type QueryDiscoverEventsArgs = {
  userId: Scalars['Int'];
};

export type QueryEventArgs = {
  id: Scalars['Int'];
};

export type QueryHostEventsArgs = {
  userId: Scalars['Int'];
};

export type QueryParticipatingEventsArgs = {
  userId: Scalars['Int'];
};

export type QueryUserArgs = {
  id: Scalars['Int'];
};

export enum RegistrationMethod {
  Facebook = 'FACEBOOK',
  Google = 'GOOGLE',
  Local = 'LOCAL',
  Twitter = 'TWITTER',
}

export type UpdateEventInput = {
  description?: Maybe<Scalars['String']>;
  endDate?: Maybe<Scalars['DateTime']>;
  id: Scalars['ID'];
  restriction?: Maybe<Scalars['Int']>;
  startDate?: Maybe<Scalars['DateTime']>;
  title?: Maybe<Scalars['String']>;
};

export type UpdateUserInput = {
  dob?: Maybe<Scalars['DateTime']>;
  email?: Maybe<Scalars['String']>;
  id: Scalars['ID'];
  name?: Maybe<Scalars['String']>;
};

export type User = {
  __typename?: 'User';
  description: Scalars['String'];
  dob: Scalars['DateTime'];
  email: Scalars['String'];
  friends: Array<User>;
  id: Scalars['ID'];
  name: Scalars['String'];
  password?: Maybe<Scalars['String']>;
  registrationMethod: RegistrationMethod;
  timezone: Scalars['String'];
};

export type CreateEventMutationVariables = Exact<{
  createEventInput: CreateEventInput;
}>;

export type CreateEventMutation = { __typename?: 'Mutation' } & {
  createEvent: { __typename?: 'Event' } & Pick<Event, 'id'>;
};

export type UploadFileMutationVariables = Exact<{
  file: Scalars['Upload'];
}>;

export type UploadFileMutation = { __typename?: 'Mutation' } & {
  uploadFile: { __typename?: 'Media' } & Pick<Media, 'id' | 'mediaType'>;
};

export type FindSpecificEventQueryVariables = Exact<{
  eventId: Scalars['Int'];
}>;

export type FindSpecificEventQuery = { __typename?: 'Query' } & {
  event: { __typename?: 'Event' } & Pick<
    Event,
    'id' | 'title' | 'description' | 'startDate' | 'endDate' | 'restriction'
  >;
};

export type DiscoverEventsQueryVariables = Exact<{
  discoverEventsUserId: Scalars['Int'];
}>;

export type DiscoverEventsQuery = { __typename?: 'Query' } & {
  discoverEvents: Array<
    { __typename?: 'Event' } & Pick<Event, 'id' | 'title' | 'description' | 'startDate' | 'endDate'>
  >;
};

export type GetUsersEventsQueryVariables = Exact<{
  UserId: Scalars['Int'];
}>;

export type GetUsersEventsQuery = { __typename?: 'Query' } & {
  coHostEvents: Array<
    { __typename?: 'Event' } & Pick<Event, 'id' | 'title' | 'description' | 'startDate' | 'endDate'>
  >;
  hostEvents: Array<
    { __typename?: 'Event' } & Pick<Event, 'id' | 'title' | 'description' | 'startDate' | 'endDate'>
  >;
};

export type GetParticipatingEventsQueryVariables = Exact<{
  participatingEventsUserId: Scalars['Int'];
}>;

export type GetParticipatingEventsQuery = { __typename?: 'Query' } & {
  participatingEvents: Array<
    { __typename?: 'Event' } & Pick<Event, 'id' | 'endDate' | 'description' | 'startDate' | 'title'>
  >;
};

export const CreateEventDocument = gql`
  mutation CreateEvent($createEventInput: CreateEventInput!) {
    createEvent(createEventInput: $createEventInput) {
      id
    }
  }
`;
export type CreateEventMutationFn = Apollo.MutationFunction<
  CreateEventMutation,
  CreateEventMutationVariables
>;

/**
 * __useCreateEventMutation__
 *
 * To run a mutation, you first call `useCreateEventMutation` within a React component and pass it any options that fit your needs.
 * When your component renders, `useCreateEventMutation` returns a tuple that includes:
 * - A mutate function that you can call at any time to execute the mutation
 * - An object with fields that represent the current status of the mutation's execution
 *
 * @param baseOptions options that will be passed into the mutation, supported options are listed on: https://www.apollographql.com/docs/react/api/react-hooks/#options-2;
 *
 * @example
 * const [createEventMutation, { data, loading, error }] = useCreateEventMutation({
 *   variables: {
 *      createEventInput: // value for 'createEventInput'
 *   },
 * });
 */
export function useCreateEventMutation(
  baseOptions?: Apollo.MutationHookOptions<CreateEventMutation, CreateEventMutationVariables>
) {
  return Apollo.useMutation<CreateEventMutation, CreateEventMutationVariables>(
    CreateEventDocument,
    baseOptions
  );
}
export type CreateEventMutationHookResult = ReturnType<typeof useCreateEventMutation>;
export type CreateEventMutationResult = Apollo.MutationResult<CreateEventMutation>;
export type CreateEventMutationOptions = Apollo.BaseMutationOptions<
  CreateEventMutation,
  CreateEventMutationVariables
>;
export const UploadFileDocument = gql`
  mutation UploadFile($file: Upload!) {
    uploadFile(file: $file) {
      id
      mediaType
    }
  }
`;
export type UploadFileMutationFn = Apollo.MutationFunction<
  UploadFileMutation,
  UploadFileMutationVariables
>;

/**
 * __useUploadFileMutation__
 *
 * To run a mutation, you first call `useUploadFileMutation` within a React component and pass it any options that fit your needs.
 * When your component renders, `useUploadFileMutation` returns a tuple that includes:
 * - A mutate function that you can call at any time to execute the mutation
 * - An object with fields that represent the current status of the mutation's execution
 *
 * @param baseOptions options that will be passed into the mutation, supported options are listed on: https://www.apollographql.com/docs/react/api/react-hooks/#options-2;
 *
 * @example
 * const [uploadFileMutation, { data, loading, error }] = useUploadFileMutation({
 *   variables: {
 *      file: // value for 'file'
 *   },
 * });
 */
export function useUploadFileMutation(
  baseOptions?: Apollo.MutationHookOptions<UploadFileMutation, UploadFileMutationVariables>
) {
  return Apollo.useMutation<UploadFileMutation, UploadFileMutationVariables>(
    UploadFileDocument,
    baseOptions
  );
}
export type UploadFileMutationHookResult = ReturnType<typeof useUploadFileMutation>;
export type UploadFileMutationResult = Apollo.MutationResult<UploadFileMutation>;
export type UploadFileMutationOptions = Apollo.BaseMutationOptions<
  UploadFileMutation,
  UploadFileMutationVariables
>;
export const FindSpecificEventDocument = gql`
  query FindSpecificEvent($eventId: Int!) {
    event(id: $eventId) {
      id
      title
      description
      startDate
      endDate
      restriction
    }
  }
`;

/**
 * __useFindSpecificEventQuery__
 *
 * To run a query within a React component, call `useFindSpecificEventQuery` and pass it any options that fit your needs.
 * When your component renders, `useFindSpecificEventQuery` returns an object from Apollo Client that contains loading, error, and data properties
 * you can use to render your UI.
 *
 * @param baseOptions options that will be passed into the query, supported options are listed on: https://www.apollographql.com/docs/react/api/react-hooks/#options;
 *
 * @example
 * const { data, loading, error } = useFindSpecificEventQuery({
 *   variables: {
 *      eventId: // value for 'eventId'
 *   },
 * });
 */
export function useFindSpecificEventQuery(
  baseOptions: Apollo.QueryHookOptions<FindSpecificEventQuery, FindSpecificEventQueryVariables>
) {
  return Apollo.useQuery<FindSpecificEventQuery, FindSpecificEventQueryVariables>(
    FindSpecificEventDocument,
    baseOptions
  );
}
export function useFindSpecificEventLazyQuery(
  baseOptions?: Apollo.LazyQueryHookOptions<FindSpecificEventQuery, FindSpecificEventQueryVariables>
) {
  return Apollo.useLazyQuery<FindSpecificEventQuery, FindSpecificEventQueryVariables>(
    FindSpecificEventDocument,
    baseOptions
  );
}
export type FindSpecificEventQueryHookResult = ReturnType<typeof useFindSpecificEventQuery>;
export type FindSpecificEventLazyQueryHookResult = ReturnType<typeof useFindSpecificEventLazyQuery>;
export type FindSpecificEventQueryResult = Apollo.QueryResult<
  FindSpecificEventQuery,
  FindSpecificEventQueryVariables
>;
export const DiscoverEventsDocument = gql`
  query DiscoverEvents($discoverEventsUserId: Int!) {
    discoverEvents(userId: $discoverEventsUserId) {
      id
      title
      description
      startDate
      endDate
    }
  }
`;

/**
 * __useDiscoverEventsQuery__
 *
 * To run a query within a React component, call `useDiscoverEventsQuery` and pass it any options that fit your needs.
 * When your component renders, `useDiscoverEventsQuery` returns an object from Apollo Client that contains loading, error, and data properties
 * you can use to render your UI.
 *
 * @param baseOptions options that will be passed into the query, supported options are listed on: https://www.apollographql.com/docs/react/api/react-hooks/#options;
 *
 * @example
 * const { data, loading, error } = useDiscoverEventsQuery({
 *   variables: {
 *      discoverEventsUserId: // value for 'discoverEventsUserId'
 *   },
 * });
 */
export function useDiscoverEventsQuery(
  baseOptions: Apollo.QueryHookOptions<DiscoverEventsQuery, DiscoverEventsQueryVariables>
) {
  return Apollo.useQuery<DiscoverEventsQuery, DiscoverEventsQueryVariables>(
    DiscoverEventsDocument,
    baseOptions
  );
}
export function useDiscoverEventsLazyQuery(
  baseOptions?: Apollo.LazyQueryHookOptions<DiscoverEventsQuery, DiscoverEventsQueryVariables>
) {
  return Apollo.useLazyQuery<DiscoverEventsQuery, DiscoverEventsQueryVariables>(
    DiscoverEventsDocument,
    baseOptions
  );
}
export type DiscoverEventsQueryHookResult = ReturnType<typeof useDiscoverEventsQuery>;
export type DiscoverEventsLazyQueryHookResult = ReturnType<typeof useDiscoverEventsLazyQuery>;
export type DiscoverEventsQueryResult = Apollo.QueryResult<
  DiscoverEventsQuery,
  DiscoverEventsQueryVariables
>;
export const GetUsersEventsDocument = gql`
  query GetUsersEvents($UserId: Int!) {
    coHostEvents(userId: $UserId) {
      id
      title
      description
      startDate
      endDate
    }
    hostEvents(userId: $UserId) {
      id
      title
      description
      startDate
      endDate
    }
  }
`;

/**
 * __useGetUsersEventsQuery__
 *
 * To run a query within a React component, call `useGetUsersEventsQuery` and pass it any options that fit your needs.
 * When your component renders, `useGetUsersEventsQuery` returns an object from Apollo Client that contains loading, error, and data properties
 * you can use to render your UI.
 *
 * @param baseOptions options that will be passed into the query, supported options are listed on: https://www.apollographql.com/docs/react/api/react-hooks/#options;
 *
 * @example
 * const { data, loading, error } = useGetUsersEventsQuery({
 *   variables: {
 *      UserId: // value for 'UserId'
 *   },
 * });
 */
export function useGetUsersEventsQuery(
  baseOptions: Apollo.QueryHookOptions<GetUsersEventsQuery, GetUsersEventsQueryVariables>
) {
  return Apollo.useQuery<GetUsersEventsQuery, GetUsersEventsQueryVariables>(
    GetUsersEventsDocument,
    baseOptions
  );
}
export function useGetUsersEventsLazyQuery(
  baseOptions?: Apollo.LazyQueryHookOptions<GetUsersEventsQuery, GetUsersEventsQueryVariables>
) {
  return Apollo.useLazyQuery<GetUsersEventsQuery, GetUsersEventsQueryVariables>(
    GetUsersEventsDocument,
    baseOptions
  );
}
export type GetUsersEventsQueryHookResult = ReturnType<typeof useGetUsersEventsQuery>;
export type GetUsersEventsLazyQueryHookResult = ReturnType<typeof useGetUsersEventsLazyQuery>;
export type GetUsersEventsQueryResult = Apollo.QueryResult<
  GetUsersEventsQuery,
  GetUsersEventsQueryVariables
>;
export const GetParticipatingEventsDocument = gql`
  query GetParticipatingEvents($participatingEventsUserId: Int!) {
    participatingEvents(userId: $participatingEventsUserId) {
      id
      endDate
      description
      startDate
      title
    }
  }
`;

/**
 * __useGetParticipatingEventsQuery__
 *
 * To run a query within a React component, call `useGetParticipatingEventsQuery` and pass it any options that fit your needs.
 * When your component renders, `useGetParticipatingEventsQuery` returns an object from Apollo Client that contains loading, error, and data properties
 * you can use to render your UI.
 *
 * @param baseOptions options that will be passed into the query, supported options are listed on: https://www.apollographql.com/docs/react/api/react-hooks/#options;
 *
 * @example
 * const { data, loading, error } = useGetParticipatingEventsQuery({
 *   variables: {
 *      participatingEventsUserId: // value for 'participatingEventsUserId'
 *   },
 * });
 */
export function useGetParticipatingEventsQuery(
  baseOptions: Apollo.QueryHookOptions<
    GetParticipatingEventsQuery,
    GetParticipatingEventsQueryVariables
  >
) {
  return Apollo.useQuery<GetParticipatingEventsQuery, GetParticipatingEventsQueryVariables>(
    GetParticipatingEventsDocument,
    baseOptions
  );
}
export function useGetParticipatingEventsLazyQuery(
  baseOptions?: Apollo.LazyQueryHookOptions<
    GetParticipatingEventsQuery,
    GetParticipatingEventsQueryVariables
  >
) {
  return Apollo.useLazyQuery<GetParticipatingEventsQuery, GetParticipatingEventsQueryVariables>(
    GetParticipatingEventsDocument,
    baseOptions
  );
}
export type GetParticipatingEventsQueryHookResult = ReturnType<
  typeof useGetParticipatingEventsQuery
>;
export type GetParticipatingEventsLazyQueryHookResult = ReturnType<
  typeof useGetParticipatingEventsLazyQuery
>;
export type GetParticipatingEventsQueryResult = Apollo.QueryResult<
  GetParticipatingEventsQuery,
  GetParticipatingEventsQueryVariables
>;
