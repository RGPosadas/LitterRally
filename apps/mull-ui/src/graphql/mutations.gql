# Creating Events
mutation CreateEvent($event: CreateEventInput!) {
  createEvent(event: $event) {
    id
  }
}

# Uploading media files
mutation UploadFile($file: Upload!) {
  uploadFile(file: $file) {
    id
    mediaType
  }
}

# Register a user
mutation CreateUser($user: CreateUserInput!) {
  createUser(user: $user) {
    id
  }
}

# Login a user
mutation Login($loginInput: LoginInput!) {
  login(loginInput: $loginInput) {
    accessToken
  }
# User joins an event
mutation JoinEvent($eventId: Int!, $userId: Int!) {
  addParticipantToEvent(eventId: $eventId, userId: $userId)
}

# User leaves an event
mutation LeaveEvent($eventId: Int!, $userId: Int!) {
  removeParticipantFromEvent(eventId: $eventId, userId: $userId)
}
