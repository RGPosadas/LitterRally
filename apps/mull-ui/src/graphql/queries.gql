fragment EventPageContent on Event {
  id
  title
  description
  startDate
  endDate
  restriction
  location {
    title
  }
  image {
    id
    mediaType
  }
  host {
    id
    name
  }
}

fragment EventCardContent on Event {
  id
  title
  restriction
  startDate
  location {
    title
  }
  image {
    id
    mediaType
  }
}

query DiscoverEvents {
  discoverEvents {
    ...EventCardContent
  }
}

# Gets all related events to a user
query OwnedEvents {
  coHostEvents {
    id
    title
    restriction
    startDate
    location {
      title
    }
    image {
      id
      mediaType
    }
  }
  hostEvents {
    id
    title
    restriction
    startDate
    location {
      title
    }
    image {
      id
      mediaType
    }
  }
}

# Get all the events that a user is participating in
query ParticipantEvents {
  participantEvents {
    id
    title
    restriction
    startDate
    location {
      title
    }
    image {
      id
      mediaType
    }
  }
}

# Find events by Id and finds participating events of user
query EventPage($eventId: Int!) {
  isParticipant(eventId: $eventId)

  event(id: $eventId) {
    ...EventPageContent
  }
}

query User {
  user {
    id
    name
    description
    avatar {
      id
    }
  }
}

query UserProfile {
  user {
    name
    description
    joinDate
    avatar {
      id
    }
  }

  friendCount
  hostingCount
  portfolioCount
}
