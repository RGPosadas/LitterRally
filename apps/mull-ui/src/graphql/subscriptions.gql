subscription PostAdded($channelId: Int!) {
  postAdded(channelId: $channelId) {
    id
    createdTime
    message
    user {
      id
      name
      avatar {
        id
        mediaType
      }
    }
  }
}
