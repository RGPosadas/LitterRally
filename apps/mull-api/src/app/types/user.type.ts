import { Field, ID, ObjectType } from '@nestjs/graphql';

@ObjectType('User')
export class UserType {
  @Field(/* istanbul ignore next */ () => ID)
  id: number;

  @Field()
  password: string;

  @Field()
  email: string;

  @Field()
  timezone: string;

  @Field()
  name: string;

  @Field()
  age: string;

  @Field()
  description: string;

  @Field(/* istanbul ignore next */ () => [UserType])
  friends: UserType[];

  // TODO: Implement modules, services, types for Entities below

  //   @Field(() => [EventType])
  //   events: EventType[];

  //   @Field(() => [PostReactionType])
  //   postReactions: PostReactionType[];

  //   @Field(() => MediaType)
  //   avatar: MediaType;
}
