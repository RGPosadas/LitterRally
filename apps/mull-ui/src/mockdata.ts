import { DetectionResult, EventRestriction, ISerializedEvent, ISerializedPost } from '@mull/types';

// TODO delete when US 2.1 is completed
export const dummyProfilePictures: string[] = [
  'https://blog.photofeeler.com/wp-content/uploads/2017/04/are-bumble-profiles-fake-how-many.jpeg',
  'https://i.pinimg.com/236x/92/f0/ed/92f0edd9b0ecefdd5b7a48b8e1f7d340.jpg',
  'https://pbs.twimg.com/profile_images/748593045566853124/9DDVz0uT_400x400.jpg',
];

export const dummyEvent: ISerializedEvent = {
  id: 1,
  title: 'Test title',
  description:
    'Lorem ipsum dolor sit amet, regione invidunt democritum vim in, movet antiopam gubergren ne per. Est nibh magna scribentur ea, vel te munere deseruisse disputationi, eros meis ludus ne sea. Odio prompta legendos in sea, ut mei scripta labores theophrastus, id molestie probatus periculis mea. Noluisse invenire splendide sed ne, at erat quando laudem nec, possim apeirian vix at. At eam animal efficiendi interpretaris, eirmod offendit adversarium per et, summo qualisque efficiendi in has. Vel detraxit accusata ea.',
  startDate: '2009-06-15T13:45:30',
  endDate: '2009-06-15T13:45:30',
  location: {
    id: 1,
    title: '1260 Remembrance Road Montreal, Qc',
  },
  restriction: EventRestriction.NONE,
  participants: [
    {
      id: 1,
      name: 'User 1',
    },
    {
      id: 2,
      name: 'User 2',
    },
  ],
  host: {
    id: 3,
    name: 'Test user',
  },
  coHosts: [
    {
      id: 4,
      name: 'First CoHost',
    },
  ],
  image: {
    id: 1,
    mediaType: 'jpeg',
  },
};

export const dummyDetectionResults: DetectionResult[] = [
  {
    bndBox: {
      height: 50,
      width: 100,
      x: 20,
      y: 40,
    },
    class: 'bottle',
    confidence: 0.8,
  },
];

const mockUser1 = {
  id: 1,
  password: '$2b$10$Wo/ClOmaI/qItblWM1saEeWgqv9S9nt6QO4KW1kBXdzPT1.q8XuuG',
  email: 'mock@mock.com',
  timezone: '',
  name: 'Bob',
  dob: new Date(),
  description: 'I am very good programmer',
  avatar: { id: 1, mediaType: 'png' },
  friends: null,
  tokenVersion: 0,
  registrationMethod: null,
  joinDate: new Date('2020-10-31T22:09:00.000Z'),
};

const mockUser2 = {
  id: 2,
  password: '$2b$10$Wo/ClOmaI/qItblWM1saEeWgqv9S9nt6QO4KW1kBXdzPT1.q8XuuG',
  email: 'mock@mock.com',
  timezone: '',
  name: 'Bob',
  dob: new Date(),
  description: 'I am very good programmer',
  avatar: { id: 1, mediaType: 'png' },
  friends: null,
  tokenVersion: 0,
  registrationMethod: null,
  joinDate: new Date('2020-10-31T22:09:00.000Z'),
};

const mockChannel = {
  id: 3,
  name: 'announcements',
  rights: 0,
  posts: null,
  participants: null,
};

export const mockPosts: ISerializedPost[] = [
  {
    parentPost: null,
    id: 1,
    channel: mockChannel,
    user: mockUser1,
    message: 'message 1 from user 1, current user id is 1',
    createdTime: '2020-10-31T22:09:00.000Z',
    media: null,
  },
  {
    parentPost: null,
    id: 2,
    channel: mockChannel,
    user: mockUser2,
    message: 'message 2 from user 2',
    createdTime: '2020-10-31T22:09:00.000Z',
    media: null,
  },
  {
    parentPost: null,
    id: 3,
    channel: mockChannel,
    user: mockUser2,
    message: 'message 3 from user 3',
    createdTime: '2020-10-31T22:09:00.000Z',
    media: null,
  },
  {
    parentPost: null,
    id: 4,
    channel: mockChannel,
    user: mockUser2,
    message: '',
    createdTime: '2020-10-31T22:09:00.000Z',
    media: {
      id: 1,
      mediaType: 'jpeg',
    },
  },
  {
    parentPost: null,
    id: 5,
    channel: mockChannel,
    user: mockUser2,
    message: 'message 4 from user 4 with a media',
    createdTime: '2020-10-31T22:09:00.000Z',
    media: {
      id: 1,
      mediaType: 'png',
    },
  },
];

export const mockChannelWithPosts = {
  id: 3,
  name: 'announcements',
  rights: 0,
  posts: mockPosts,
  event: dummyEvent,
};
