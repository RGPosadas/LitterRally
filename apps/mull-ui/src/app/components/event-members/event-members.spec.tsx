import { MockedProvider, MockedResponse } from '@apollo/client/testing';
import { ISerializedEvent } from '@mull/types';
import { GetThreeRandomParticipantsDocument } from '../../../generated/graphql';
import React from 'react';
import renderer, { act as actRenderer } from 'react-test-renderer';
import { EventMembers } from '..';
import { dummyEvent } from '../../../mockdata';

describe('EventMembers', () => {
  const renderHelper = (mocks: MockedResponse[], event: ISerializedEvent) => {
    return (
      <MockedProvider mocks={mocks}>
        <EventMembers eventId={event.id} />
      </MockedProvider>
    );
  };

  it('should render successfully', async () => {
    const mocks: MockedResponse[] = [
      {
        request: {
          query: GetThreeRandomParticipantsDocument,
          variables: { eventId: dummyEvent.id },
        },
        result: {
          data: { participants: dummyEvent.participants },
        },
      },
    ];

    await actRenderer(async () => {
      const tree = renderer.create(renderHelper(mocks, dummyEvent));
      await new Promise((resolve) => setTimeout(resolve, 200));
      expect(tree.toJSON()).toMatchSnapshot();
    });
  });
});
