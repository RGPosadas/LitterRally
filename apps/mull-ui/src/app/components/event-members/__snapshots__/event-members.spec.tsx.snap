// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventMembers should render successfully 1`] = `null`;
